pandas/core/indexes/base.py

```diff
@@ -96,6 +96,7 @@
     duplicated="np.ndarray",
 )
 _index_shared_docs = dict()
+str_t = str
```

Comment:
why are you doing this?

Reply:
mypy wont allow us to annotate with "str" because Index has a "str" attribute.  Same thing with how we use "bool_t" in generic.py

Reply:
ok, we should simply move these to _typing then (can be a followon)

Reply:
and we should change the str attribute to something else

Reply:
i think we're OK with this being a follow-up, yah?